### ISSUE TITLE
[Bug]: Remix Dev Server Process not terminated once Remix Application is set up with Prisma

### ISSUE THREAD

**Issue by dggsax** [remix-run/remix#762]
### Which Remix packages are impacted?

- [ ] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.0.6

## Steps to introduce the issue
_The following steps were done on two separate windows devices. I have not attempted this on a Mac device (yet)._
1. Create a new Remix project using `npx create-remix@latest`
2. Follow the steps in the Jokes tutorial to [Set up Prisma](https://remix.run/docs/en/v1/tutorials/jokes#set-up-prisma) and [Connect to the database](https://remix.run/docs/en/v1/tutorials/jokes#connect-to-the-database).
3. Load some data in one of the existing project pages so that when you run `npm run dev` we establish a connection with the Prisma DB
```diff
# inside app\routes\demos\about.tsx
...
export let links: LinksFunction = () => {
  return [{ rel: "stylesheet", href: stylesUrl }];
};

+export let loader: LoaderFunction = async (): Promise<Joke[]> => {
+  return await db.joke.findMany();
+}

export default function Index() {
+  const jokes = useLoaderData<Joke[]>();

  return (
    <div className="about">
      <div className="about__intro">
        <h2>About Us</h2>
+        {
+          jokes.map(j => <p key={j.id}>{j.name}</p>)
+        }
        <p>
          Ok, so this page isn't really <em>about us</em>, but we did want to
          show you a few more things Remix can do.
        </p>
...
```
4. Run `npm run dev` and navigate to http://localhost:3000/demos/about so we load the data from Prisma and see our Joke names.
5. Execute `Ctrl + C` to terminate the dev server _(scenario: say you needed to install an NPM package or something)_
6. Attempt to run `npm run dev` again and you receive the following error:

```
PS D:\tmp\remix-repro> npm run dev

> dev
> remix dev

Watching Remix app in development mode...
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::8002
    at Server.setupListenHandle [as _listen2] (net.js:1318:16)
    at listenInCluster (net.js:1366:12)
    at Server.listen (net.js:1452:7)
    at new WebSocketServer (D:\tmp\remix-repro\node_modules\ws\lib\websocket-server.js:90:20)
    at watch (D:\tmp\remix-repro\node_modules\@remix-run\dev\cli\commands.js:89:13)
    at Object.dev (D:\tmp\remix-repro\node_modules\@remix-run\dev\cli\commands.js:182:11)
Emitted 'error' event on WebSocketServer instance at:
    at Server.emit (events.js:315:20)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  code: 'EADDRINUSE',
  errno: -4091,
  syscall: 'listen',
  address: '::',
  port: 8002
}
```

## Steps the "resolve" the issue
_I'm putting this here because it's helped me to narrow it down to be something related to initiating the DB connection._
 7. (Continuing where we left off) The Remix Dev Server is still running on port 8002 so we need to terminate it. Since I'm on Windows, I do this with `Get-Process -Id (Get-NetTCPConnection -LocalPort 8002).OwningProcess | Stop-Process`
 8. Comment out the code we added to `app\routes\demos\about.tsx` so that none of the files in our Remix project establish a connection with the DB.
 9. Run `npm run dev` to re-start the dev server. Run `Ctrl + C` to terminate the dev server.
 10. Run `npm run dev` one more time. This time, you won't get the  `Error: listen EADDRINUSE: address already in use :::8002` error from above.

### Expected Behavior

Once I've reached the part of the [Jokes App Tutorial](https://remix.run/docs/en/v1/tutorials/jokes) where we've set up Prisma and have started interacting with the Data for the tutorial, I expect to be able to run `npm run dev`, terminate that process with `Ctrl + C` when I need to do something like install an NPM package, and then get back to work with `npm run dev`.

### Actual Behavior

As described in the Steps to Reproduce, once I've set up the starter Remix project with Prisma and added a simple loader to display data from the DB in the UI, the Remix Dev Server does not terminate upon `Ctrl + C`. As a result, when I try to run `npm run dev` in the same terminal/PowerShell process, the Dev Server is unable to start since the old process is still running and am hit with the following error (pasting again for convenience):
```
PS D:\tmp\remix-repro> npm run dev

> dev
> remix dev

Watching Remix app in development mode...
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::8002
    at Server.setupListenHandle [as _listen2] (net.js:1318:16)
    at listenInCluster (net.js:1366:12)
    at Server.listen (net.js:1452:7)
    at new WebSocketServer (D:\tmp\remix-repro\node_modules\ws\lib\websocket-server.js:90:20)
    at watch (D:\tmp\remix-repro\node_modules\@remix-run\dev\cli\commands.js:89:13)
    at Object.dev (D:\tmp\remix-repro\node_modules\@remix-run\dev\cli\commands.js:182:11)
Emitted 'error' event on WebSocketServer instance at:
    at Server.emit (events.js:315:20)
    at emitErrorNT (net.js:1345:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  code: 'EADDRINUSE',
  errno: -4091,
  syscall: 'listen',
  address: '::',
  port: 8002
}
```
I've been getting around this by mapping `Get-Process -Id (Get-NetTCPConnection -LocalPort 8002).OwningProcess | Stop-Process` to a PowerShell Alias. Before I was having to start a new instance of Windows Terminal every time, which was getting inconvenient.

One user in the Remix Discord has mentioned running into this issue on their Macbook: "just got the same error as you, I am on a MB Pro M1 with macOS 12.0.1, within VS Code terminal integration"

**Comment by xstevenyung** [comment]
Can confirm, got the same issue on MBP (Intel), as a quick fix it's possible on linux based OS to use `ps` to list process and manually kill the remix process still running.

Result of `ps`:
<img width="1128" alt="Screenshot 2021-12-04 at 12 24 08" src="https://user-images.githubusercontent.com/7261258/144707642-4a106c26-d11e-4dc2-9d47-456c94aa6588.png">

Then you can kill the process with `kill 3972`.

**Comment by machour** [comment]
Fixed in #752
